jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/admin/commands/TriggerNodeCommand.java

```diff
@@ -43,28 +44,40 @@ public TriggerNodeCommand(long processInstanceId, long nodeId) {
 
     public Void execute(Context context) {
     	KieSession kieSession = ((RegistryContext) context).lookup( KieSession.class );
-    	logger.debug("About to trigger (create) node instance for node {} in process instance {}", nodeId, processInstanceId); 
+    	logger.debug("About to trigger (create) node instance for node {} in process instance {}", nodeId, processInstanceId);
         RuleFlowProcessInstance wfp = (RuleFlowProcessInstance) kieSession.getProcessInstance(processInstanceId, false);
         if (wfp == null) {
             throw new ProcessInstanceNotFoundException("Process instance with id " + processInstanceId + " not found");
         }
-        
-        Node node = wfp.getRuleFlowProcess().getNodesRecursively().stream().filter(ni -> ni.getId() == nodeId).findFirst().orElse(null);
-        if (node == null) {
-            throw new NodeNotFoundException("Node instance with id " + nodeId + " not found");
+
+        Node node = null;
+        try {
+            node = wfp.getRuleFlowProcess().getNode(nodeId);
+        } catch (Exception e) {
+
+        }
+
+        if (node == null) { //search container node
+            node = wfp.getRuleFlowProcess().getNodesRecursively().stream().filter(ni -> ni.getId() == nodeId).findFirst().orElse(null);
+            if (node == null) {
+                throw new NodeNotFoundException("Node instance with id " + nodeId + " not found");
+            }
+            CompositeNodeInstance nodeInstanceContainerNode =
+                    ((CompositeNodeInstance) wfp.getNodeInstance(wfp.getRuleFlowProcess().getParentNode(nodeId)));
```

Comment:
we sure it's always CompositeNodeInstance? or shall we make it NodeContainer as the variable name suggests?

Reply:
@mswiderski  Nop, you are absolutely right I changed that to a generic NodeInstanceContainer and it works in the same way. Better now, before it was to tied to the examples I tried, Thanks!